### PR TITLE
added support to pass modules from container app to child

### PR DESCRIPTION
**Pull request by shyoro**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -42,46 +42,87 @@ module.exports.pitch = function (remainingRequest) {
...
+    globalVar = `${this.query.namespace.replace(/^\?/i, '')}.${request}`;
+  } else { //Use modules from parent app
+    request = request[request.length - 1].replace(/^./i, '').replace(/\//g, '.');
+    globalVar = `${this.query.namespace}${request}`;
+  }
+
   this._module.userRequest = this._module.userRequest + '-shared';
-  return accesorString(globalVar) + " = " +
+
+  const result = accesorString(globalVar) + " = " +
```
Comment (MrFrankel) [inline index.js]: redundant const, can return immediately

File: index.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ var path = require('path');
 function accesorString(value) {
   const childProperties = value.split(".");
   const length = childProperties.length;
-  let propertyString = "window";
+  let propertyString = "global";
```
Comment (MrFrankel) [inline index.js]: better use window, some cases global is not defined

Comment (MrFrankel) [reply index.js]: better use window, some cases global is not defined

File: index.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ function accesorString(value) {
 function propertyString(value) {
   const childProperties = value.split(".");
   const length = childProperties.length;
-  let propertyString = "window";
+  let propertyString = "global";
```
Comment (MrFrankel) [inline index.js]: better use window, some cases global is not defined

File: index.js [hunk 4]
```diff
@@ -42,46 +42,87 @@ module.exports.pitch = function (remainingRequest) {
   // This prevents [chunkhash] values from changing when running webpack
   // builds in different directories.
   // this.loadModule('@angular/core', (a,b,c,d) =>{debugger;});
-  if (this.query.exclude.some(mdl => this._module.rawRequest.match(new RegExp(mdl)))){
```
Comment (MrFrankel) [inline index.js]: why is this removed? it's important for excluding modules from being exposed

File: index.js [hunk 5]
```diff
@@ -42,46 +42,87 @@ module.exports.pitch = function (remainingRequest) {
...
   let request = this._module.rawRequest.split('!');
-  request = request[request.length - 1].replace(/^@/i, '').replace(/\//g, '.');
-  const globalVar = `${this.query.namespace.replace(/^\?/i, '')}.${request}`;
+
+  if (this._module.userRequest.includes('/node_modules/')) {
+    request = request[request.length - 1].replace(/^@/i, '').replace(/\//g, '.');
+    globalVar = `${this.query.namespace.replace(/^\?/i, '')}.${request}`;
+  } else { //Use modules from parent app
+    request = request[request.length - 1].replace(/^./i, '').replace(/\//g, '.');
+    globalVar = `${this.query.namespace}${request}`;
```
Comment (MrFrankel) [inline index.js]: should this be: 
`
const globalVar = `${this.query.namespace.replace(/^\?/i, '')}.${request}`;
`

File: index.js [hunk 6]
```diff
@@ -42,46 +42,87 @@ module.exports.pitch = function (remainingRequest) {
...
+  let globalVar;
   let request = this._module.rawRequest.split('!');
-  request = request[request.length - 1].replace(/^@/i, '').replace(/\//g, '.');
-  const globalVar = `${this.query.namespace.replace(/^\?/i, '')}.${request}`;
+
+  if (this._module.userRequest.includes('/node_modules/')) {
+    request = request[request.length - 1].replace(/^@/i, '').replace(/\//g, '.');
+    globalVar = `${this.query.namespace.replace(/^\?/i, '')}.${request}`;
+  } else { //Use modules from parent app
+    request = request[request.length - 1].replace(/^./i, '').replace(/\//g, '.');
```
Comment (MrFrankel) [inline index.js]: should this: `replace(/^./i, '')`

be 
replace(/^@/i, '')
``

**Comment by MrFrankel** [comment]
do you need an npm release?

File: index.js (hunk 4 above)
Comment (shyoro) [reply index.js]: fixed, added check for exclude object

**Comment by shyoro** [comment]
yes it would be great if you can release to npm

**Comment by MrFrankel** [comment]
releasd in version 6.0.0
